src/plot_api/subroutines.js

```diff
@@ -124,31 +124,20 @@ exports.lsInner = function(gd) {
 
         var xDomain = plotinfo.xaxis.domain;
         var yDomain = plotinfo.yaxis.domain;
-        var plotgroupBgData = [];
+        var plotgroup = plotinfo.plotgroup;
+        var plotgroupBg;
 
         if(overlappingDomain(xDomain, yDomain, lowerDomains)) {
-            plotgroupBgData = [0];
-        }
-        else {
+            var pgNode = plotgroup.node();
+            plotgroupBg = plotinfo.bg = Lib.ensureSingle(plotgroup, 'rect', 'bg');
+            pgNode.insertBefore(plotgroupBg.node(), pgNode.childNodes[0]);
+        } else {
+            plotgroupBg = plotgroup.select('rect,bg');
+            if(plotgroupBg.size()) plotgroupBg.remove();
```

Comment:
does this `if` make a difference? Does it take a non-negligible time to `remove` an empty selection?

Reply:
that `if` did not improve things -> :hocho: in 5307125ef9655bec0c2e24ba444b3cfc4c1c4f48